pysr/test/test.py

```diff
@@ -513,6 +513,282 @@ def test_jl_function_error(self):
+            X.append(X[i - 1] + X[i - 2] + X[i - 3])
+        X = np.asarray(X).reshape(-1, 1)
+        model = PySRSequenceRegressor(
+            **self.default_test_kwargs,
+            binary_operators=["+"],
+            early_stop_condition="stop_if(loss, complexity) = loss < 1e-4 && complexity == 1",
+        )
+        model.fit(X)
+        print(model.equations_)
+        self.assertLessEqual(model.get_best()["loss"], 1e-4)
+
+    def test_sequence_named(self):
+        X = [1, 1, 1]
+        for i in range(3, 30):
+            X.append(X[i - 1] + X[i - 2] + X[i - 3])
+        X = np.asarray(X).reshape(-1, 1)
+        model = PySRSequenceRegressor(
+            **self.default_test_kwargs,
+            early_stop_condition="stop_if(loss, complexity) = loss < 1e-4 && complexity == 1",
+        )
+        model.fit(X, variable_names=["c1"])
+        self.assertIn("c1t_1", model.equations_.iloc[-1]["equation"])
+
+    def test_sequence_weighted_bumper(self):
+        X = [1, 1, 1]
+        for i in range(3, 30):
+            X.append(X[i - 1] + X[i - 2] + X[i - 3])
+        X = np.asarray(X).reshape(-1, 1)
+        weights = np.ones_like(X).reshape(-1)
+        model = PySRSequenceRegressor(
+            **self.default_test_kwargs,
+            early_stop_condition="stop_if(loss, complexity) = loss < 1e-4 && complexity <= 5",
+            bumper=True,
+        )
+        model.fit(X, weights=weights)
+        print(model.equations_)
+        self.assertLessEqual(model.get_best()["loss"], 1e-1)
+        self.assertEqual(
+            jl.seval("((::Val{x}) where x) -> x")(model.julia_options_.bumper), True
+        )
+
+    def test_sequence_multiprocessing_turbo_custom_objective(self):
+        X = [1]
+        for i in range(1, 20):
+            X.append(np.sqrt(X[i - 1]) + 1)
+        X = np.asarray(X).reshape(-1, 1)
+        model = PySRSequenceRegressor(
+            **self.default_test_kwargs,
+            # Turbo needs to work with unsafe operators:
+            unary_operators=["sqrt"],
+            procs=2,
+            multithreading=False,
+            turbo=True,
+            early_stop_condition="stop_if(loss, complexity) = loss < 1e-10 && complexity == 1",
+            loss_function="""
+            function my_objective(tree::Node{T}, dataset::Dataset{T}, options::Options) where T
+                prediction, flag = eval_tree_array(tree, dataset.X, options)
+                !flag && return T(Inf)
+                abs3(x) = abs(x) ^ 3
+                return sum(abs3, prediction .- dataset.y) / length(prediction)
+            end
+            """,
+        )
+        model.fit(X)
+        print(model.equations_)
+        best_loss = model.equations_.iloc[-1]["loss"]
+        self.assertLessEqual(best_loss, 1e-10)
+        self.assertGreaterEqual(best_loss, 0.0)
+
+        # Test options stored:
+        self.assertEqual(
+            jl.seval("((::Val{x}) where x) -> x")(model.julia_options_.turbo), True
+        )
```

Comment:
Not sure what this test is probing differently from the one in `TestPipeline`? The sequence regressor tests should only test parts unique to the sequence regressor, rather than testing all of the options again.